static/app/components/carousel.tsx

```diff
@@ -0,0 +1,148 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconArrow} from 'sentry/icons';
+import {ArrowProps} from 'sentry/icons/iconArrow';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+type Props = {
+  children?: React.ReactNode;
+};
+
+function Carousel({children}: Props) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [anchorRefs, setAnchorRefs] = useState<HTMLElement[]>([]);
+  const [childrenRefs, setChildrenRefs] = useState<HTMLElement[]>([]);
+  const [isAtStart, setIsAtStart] = useState(true);
+  const [isAtEnd, setIsAtEnd] = useState(false);
+
+  useEffect(() => {
+    if (!ref.current) {
+      return () => {};
+    }
+
+    const observer = new IntersectionObserver(
+      entries => {
+        entries.forEach(e => {
+          if (e.target.id === anchorRefs[0].id) {
+            setIsAtStart(e.isIntersecting);
+          } else if (e.target.id === anchorRefs[1].id) {
+            setIsAtEnd(e.isIntersecting);
+          }
+        });
+      },
+      {root: ref.current, threshold: [1]}
+    );
+
+    if (anchorRefs) {
+      anchorRefs.map(anchor => observer.observe(anchor));
+    }
+
+    return () => observer.disconnect();
+  }, [anchorRefs]);
+
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    setChildrenRefs(Array.from(ref.current.children) as HTMLElement[]);
+    const anchors = [
+      ref.current.children[0],
+      ref.current.children[ref.current.children.length - 1],
+    ] as HTMLElement[];
+    setAnchorRefs(anchors);
+  }, [children]);
+
+  const handleScroll = (direction: string) => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+
+      if (direction === 'left') {
+        // scroll to the last child to the left of the left side of the container
+        const elements = childrenRefs.filter(child => child.offsetLeft < scrollLeft);
+        ref.current.scrollTo(elements[elements.length - 1].offsetLeft, 0);
+      } else if (direction === 'right') {
+        // scroll to the first child to the right of the left side of the container
+        const elements = childrenRefs.filter(child => child.offsetLeft > scrollLeft);
+        ref.current.scrollTo(elements[0].offsetLeft, 0);
+      }
+    });
+  };
+
+  const showArrows = !(isAtStart && isAtEnd);
```

Comment:
Why do we need this? If `isAtStart` and `isAtEnd` are both false, then neither arrow will be visible.